wemake_python_styleguide/visitors/ast/classes.py

```diff
@@ -120,23 +121,27 @@ def _is_correct_base_class(self, base_class: ast.AST) -> bool:
         return False
 
     def _check_getters_setters_methods(self, node: ast.ClassDef) -> None:
-        class_attributes, instance_attributes = classes.get_attributes(node)
-        attribute_names = {
-            class_attribute.lstrip('_') for class_attribute
-            in name_nodes.flat_variable_names(class_attributes)
-        }.union({
-            instance.attr.lstrip('_') for instance
-            in instance_attributes
-        })
-
-        for method_postfix in classes.getter_setter_postfixes(node):
-            if any(name == method_postfix for name in attribute_names):
-                self.add_violation(
-                    oop.UnpythonicGetterSetterViolation(
-                        node,
-                        text=node.name,
-                    ),
-                )
+        class_attributes, instance_attributes = classes.get_all_attributes_str(
+            node,
+        )
+        class_postfixes, instance_postfixes = classes.get_set_postfixes(node)
```

Comment:
So, why do we care about this difference here about `class_` and `instance_` attributes? I ask because we end up with the same logic for both of them.

Reply:
classmethods can only access class attributes while instance methods can access both and it is why we have to split the checks.

Reply:
Can you please provide an example when this is useful? I am a bit lost 🙂

Reply:
```
instance_attribute_class_getter_setter = """
class Test(object):
    def __init__(self):
        self.attribute = 1

    @classmethod
    def get_attribute(self):
        ...

    @classmethod
    def set_attribute(self):
        ...
"""
```
@classmethod decorated methods cannot access the instance attribute, so there should be no violation. But my previous code raised a violation here so i realised i needed to split up the checking.

Reply:
Ok, now I see. Thank you for the explanation. This is an interesting case indeed.

The thing about this is not that this violation can be technically correct. We care about the idealogical part here. And from the idealogical point of view: there's no difference in `@classmethod` or instance methods. They are still java-style getters and setters.

So, I guess we should move towards restricting this example as well.

And combine these two logical conditions into one: with both instance and class attributes.

Reply:
Okay, then i will change to the old code and replace the tests to be expected to raise a violation.

Reply:
This would be commit: `Refacotr: Remove decorator checking`
(6a1eccc)

Reply:
👍